ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp

```diff
@@ -588,12 +588,15 @@ const ComplexTensor& grad, const ComplexTensor& input, const ComplexTensor& othe
     return grad_tensor;
 }
 
-std::vector<Tensor> ExecuteBackwardMul::invoke(
-    const Tensor& grad, const Tensor& input, float scalar, const std::optional<MemoryConfig>& output_mem_config) {
-    std::vector<Tensor> grad_tensor;
-    Tensor result = ttnn::multiply(grad, scalar, std::nullopt, output_mem_config);
-    grad_tensor.emplace_back(result);
-    return grad_tensor;
+std::vector<std::optional<ttnn::Tensor>> ExecuteBackwardMul::invoke(
+    uint8_t queue_id, const Tensor& grad, const Tensor& input, float scalar, const std::optional<MemoryConfig>& output_mem_config, std::optional<Tensor> input_grad) {
+    std::vector<std::optional<Tensor>> result;
+     if(!input_grad.has_value()){
+        input_grad = grad;
```

Comment:
This code will overwrite the buffers in the `grad` tensor that was passed in.  So the grad is being treated as an output tensor if the input_grad was not provided but will not be if the input_grad was provided.

Reply:
Will replace it with Zeros like. Thank you.